components/mailchimp/mailchimp.app.js

```diff
@@ -0,0 +1,681 @@
+        }
+        return links.urls_clicked.map((link) => ({
+          label: link.url,
+          value: link.id,
+        }));
+      },
+    },
+    segmentId: {
+      type: "string",
+      label: "Segment/Tag Id",
+      description:
+        "The unique ID of the segment or tag you'd like to watch for new subscribers.",
+      useQuery: true,
+      async options(opts) {
+        const count = 1000;
+        const offset = 1000 * opts.page;
+        const config = {
+          count: count,
+          offset: offset,
+        };
+        const segmentsResults =  await this.getAllAudienceSegments(
+          opts.listId,
+          config,
+        );
+        return segmentsResults.segments.map((segment) => ({
+          label: `${segment.name}`,
+          value: segment.id,
+        }));
+      },
+    },
+  },
+  methods: {
+    _auths() {
+      return this.$auth;
+    },
+    _authToken() {
+      return this.$auth.oauth_access_token;
+    },
+    _isRetriableStatusCode(statusCode) {
+      [
+        408,
+        429,
+        500,
+      ].includes(statusCode);
+    },
+    _server() {
+      return this.$auth.dc;
+    },
+    api() {
+      mailchimp.setConfig({
+        accessToken: this._authToken(),
+        server: this._server(),
+      });
+      return mailchimp;
+    },
+    async _withRetries(apiCall) {
+      const retryOpts = {
+        retries: 5,
+        factor: 2,
+      };
+      return retry(async (bail) => {
+        try {
+          return await apiCall();
+        } catch (err) {
+          const { status = 500 } = err;
+          if (!this._isRetriableStatusCode(status)) {
+            bail(`
+              Unexpected error (status code: ${status}):
+              ${JSON.stringify(err.response)}
+            `);
+          }
+          console.warn(`Temporary error: ${err.message}`);
+          throw err;
+        }
+      }, retryOpts);
+    },
+    statusIsSent(status) {
+      return "sent" === status;
+    },
+    /**
+     * Create a webhook on the specified Audience List.
+     * @param {String} listId - The unique ID that identifies the Audience List you would like to
+     * create a webhook on.
+     * @param {String} config.url - URL listening for the webhook event.
+     * @param {String} config.events -
+     * @param {String} config.sources -
+     * @returns {_links: array, events: object, id: string, list_id: string, sources: object, url:
+     * string} Object with array of links for the webhook manipulation through the Mailchimp API,
+     * an object with flags of the Mailchimp Marketing indicates the events being watched for, an
+     * unique ID of the webhoom, the unique ID which identifies the Audience List where the webhoook
+     * was created, a `sources` object indicating whether the sources for the event are an admin
+     * user, an audience subscriber, or via the API, the URL registered for the webhook event.
+     */
```

Comment:
I don't think we need to document each value in `config` (url, events, sources). You could document just config and include a link to the relevant documentation. This applies to all methods with `config` as a parameter.